AddOns/BasicFinancials/app/src/codeunit/BasicFinancialsMgmt.Codeunit.al

```diff
@@ -0,0 +1,125 @@
+        UnknowUserErr: Label 'The current user is unknown. The user who deploys the Basic Financials extension must exist in the user table.';
+
+    internal procedure IsSupportedLicense(): Boolean // Microsoft requirements: The Basic Financials Assisted Setup checks for the Basic Financials license on the AAD tenant, at least one user has been assigned to this license.
+    var
+    begin
+        AzureADLicensing.ResetSubscribedSKU();
+        while AzureADLicensing.NextSubscribedSKU() do
+            case UpperCase(AzureADLicensing.SubscribedSKUId()) of
+                BFSKUIdTxt:
+                    exit(true);
+                BFC5SPLASKUIdTxt:
+                    exit(true);
+            end;
+        exit(false);
+    end;
+
+    internal procedure TestSupportedLicenses() // Microsoft requirements: The extension checks for the Basic Financials license on the AAD tenant, at least one user has been assigned to this license. 
+    var
+    begin
+        AzureADLicensing.ResetSubscribedSKU();
+        while AzureADLicensing.NextSubscribedSKU() do
+            case UpperCase(AzureADLicensing.SubscribedSKUId()) of
+                BFSKUIdTxt:
+                    exit;
+                BFC5SPLASKUIdTxt:
+                    exit;
+            end;
+        Error(NotSupportedLicenseErr);
+    end;
+
+    internal procedure TestSupportedLocales() // Microsoft requirements: The extension checks for the country availability. The extension is only available in Denmark and Iceland.;
+    var
+        TempApplicationAreaSetup: Record "Application Area Setup";
+        AppAreaMgmt: Codeunit "App Area Mgmt BF";
+
+    begin
+        Clear(TempApplicationAreaSetup);
+        AppAreaMgmt.GetEssentialExperienceAppAreas(TempApplicationAreaSetup);
+        if TempApplicationAreaSetup."Basic DK" or TempApplicationAreaSetup."Basic IS" then
+            exit;
+
+        Error(NotSupportedLocalesErr);
+    end;
+
+    internal procedure TestSupportedUser() // Microsoft requirements: The extension checks whether the Super User permission set is assigned to the user who is installing the extension.
+    var
+        User: Record User;
+        UserPermissions: Codeunit "User Permissions";
+    begin
+        if UserSecurityId() = UserSecurityIdTxt then
+            Error(NotSupportedSystemUserErr);
+
+        if User.Get(UserSecurityId()) then
+            Error(UnknowUserErr);
+
+        if UserPermissions.IsSuper(UserSecurityId()) then
+            exit;
+
+        Error(NotSupportedUserErr);
+    end;
+
+    internal procedure TestSupportedCompanies() // Microsoft requirements: The extension checks whether the tenant contains more than one company during installation, but allows additional companies to be added afterward.
+    var
+        Company: Record Company;
+    begin
+        Clear(Company);
+        if Company.count() = 1 then
```

Comment:
So, this rule is problematic from many points. Partner can create some draft company or create backup and suddenly the extension cannot be installed. Can you please clarify the requirement?

Reply:
I concur. I have previously informed about the issue. Fredrik Hietala (Microsoft) has determined this as a Microsoft requirements: The extension checks whether the tenant contains more than one company during installation, but allows additional companies to be added afterward.
I've done what I've been asked to do, I think it is up to Microsoft to decide. Please reconcile it internally in Microsoft and then notify if you want to change approach.